casadm/cas_lib.c

```diff
@@ -1831,7 +1831,7 @@ int check_if_mounted(int cache_id, int core_id)
 	while ((mstruct = getmntent(mtab)) != NULL) {
 		/* mstruct->mnt_fsname is /dev/... block device path, not a mountpoint */
 		if ((NULL != mstruct->mnt_fsname)
-		    && (strncmp(mstruct->mnt_fsname, dev_buf, dev_buf_len) == 0)) {
+		    && (strcmp(mstruct->mnt_fsname, dev_buf) == 0)) {
```

Comment:
I believe it is safer to leave `strncmp` here and add 1 to `def_buf_len` few lines above to include terminal character.
To be hypercorrect You could use there `env_strnlen` in place of actual `strnlen`.
I wrote 2 tests for case like Yours - https://github.com/Open-CAS/open-cas-linux/pull/479